pkg.install and pkg.remove fix version number input.

damon-atkins opened the pull request:
### What does this PR do?
    pkg.install and pkg.remove on the command line take version number e.g. 1.2 for software and
    store them within a float as its a valid float.  However version needs to be a string, to support versions
    numbers like 1.3.4

    The following for example would not work as version number would be a float, and a
    float does not match string e.g. 3.1 <> '3.1' .
    c:\salt\salt-call -l debug pkg.install name=softwarename version=3.1

### What issues does this PR fix or reference?

### Previous Behavior
The following
c:\salt\salt-call -l debug pkg.install name=softwarename version=3.1
would fail  i.e. `version_num (float) in pkginfo`  will not match against a string. All version numbers within pkginfo are strings so as to support version number like 1.2.3 which are not a valid float.

It seems difficult to pass version=3.1 as a string on the command line.

### New Behavior
If version number is not a string it is converted to a string. Which results in a successful match.

### Tests written?
No

Commit:
pkg.install and pkg.remove on the command line take number version numbers,
store them within a float. However version is a string, to support versions
numbers like 1.3.4

The following for example would not work as version would be a float, and a
float does not match a string a dict with key of strings.
c:\salt\salt-call -l debug pkg.install name=softwarename version=3.1

salt-jenkins commented:
@damon-atkins, thanks for your PR! By analyzing the history of the files in this pull request, we identified @twangboy, @terminalmage and @UtahDave to be potential reviewers.

Commit:
Fix typo

Commit:
Handle version=None  when converted to a string it becomes 'None'
parm should default to empty string rather than None, it would fix better with existing code.

damon-atkins commented:
I notice doing testing that pkg.install if you do not provide a version, will upgrade to the latest version. Due to these lines
```
        if not version_num:
            version_num = _get_latest_pkg_version(pkginfo)
```

twangboy commented on salt/modules/win_pkg.py:
```diff
@@ -1020,10 +1020,15 @@ def install(name=None, refresh=False, pkgs=None, **kwargs):
         if options:
-            version_num = options.get('version', False)
+            version_num = options.get('version', '')
+            #  The user user salt cmdline with version=5.3 might be interpreted
```
Need to fix the comment `the user user salt cmdline...`

twangboy commented on salt/modules/win_pkg.py:
```diff
@@ -1355,6 +1360,11 @@ def remove(name=None, pkgs=None, version=None, **kwargs):
             continue
 
         if version_num is not None:
+            #  The user user salt cmdline with version=5.3 might be interpreted
```
Same thing here

Commit:
Fix comment typo

cachedout commented:
@twangboy Re-review requested.
